ui/src/views/Rules/ListRules/index.jsx

```diff
@@ -454,28 +454,44 @@ function ListRules(props) {
         rule.product || (rule.scheduledChange && rule.scheduledChange.product);
       const ruleScheduledChangeChannel =
         rule.scheduledChange && rule.scheduledChange.channel;
-      let ruleChannel;
+      const ruleChannel = rule.channel;
 
       if (ruleProduct !== productFilter) {
         return false;
       }
 
-      if (ruleScheduledChangeChannel) {
-        if (ruleScheduledChangeChannel.includes('*')) {
-          ruleChannel = rule.channel && ruleScheduledChangeChannel;
-        } else {
-          ruleChannel = ruleScheduledChangeChannel && rule.channel;
+      if (channelFilter) {
+        // if neither the rule nor the scheduled rule's channel is an exact
+        // match for the filter (after stripping away a possible wildcard)
+        // this rule does not match
+        // similarly, if the rule or scheduled rule has a wildcard, and the
+        // selected channel does not match either, this rule does not match
+        let ruleChannelMatches = false;
+        let scChannelMatches = false;
+
+        if (ruleChannel) {
+          if (ruleChannel.indexOf('*') === -1) {
+            if (ruleChannel === channelFilter) {
+              ruleChannelMatches = true;
+            }
+          } else if (channelFilter.startsWith(ruleChannel.split('*')[0])) {
+            ruleChannelMatches = true;
+          }
```

Comment:
Also a nice shorthand is to use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (`?.`):
```js
const ruleChannelMatches = ruleChannel?.includes(channelFilter);
```

Reply:
TIL! Thank you both for your suggestions; I'll take a crack at improving this.

Reply:
I did realize that just using includes may produce false matches for things like filtering by `release` but seeing `release-sysaddon` as well. i think this is probably a better solution:

```js
const ruleChannelMatches = ruleChannel?.split("*").includes(channelFilter);
```

Reply:
This looks like, but unfortunately our build doesn't work with modern node versions right now so I had to forego it =\